### PR TITLE
refactor: docker image for Cruise Control

### PR DESCRIPTION
**Pull request by chrisgacsal**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | ST-46
| License         | Apache 2.0

### What's in this PR?
* fix `MinRAMPercentage` being set to higher value than `MinRAMPercentage` 
* set the default JAVA HEAP parameters to 75% of the available memory in `KAFKA_HEAP_OPTS`
* use AdoptOpenJDK distribution for both build and runtime images
* use JRE for runtime to reduce the size of the final image

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

**Review by hi-im-aren — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,40 +1,51 @@
...
-RUN apt-get update && apt-get install -y --no-install-recommends git ca-certificates
-RUN git clone -b ${VERSION} https://github.com/linkedin/cruise-control.git
-RUN cd cruise-control && ./gradlew jar copyDependantLibs
+RUN \
+  set -xe; \
+  apt-get update -qq \
+  && apt-get install -qq --no-install-recommends \
+    git ca-certificates
+RUN \
+  set xe; \
```
Comment (schultheiszn) [inline Dockerfile:11]: looks like a `-` is missing here

**Review by baluchicken — APPROVED** [review]
LGTM thx @chrisgacsal

File: Dockerfile (hunk 1 above)
Comment (baluchicken) [reply Dockerfile:11]: Nice catch:)

**Review by stoader — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (chrisgacsal) [reply Dockerfile]: Thanks @schultheiszn . Fixed.

File: Dockerfile [hunk 2]
```diff
@@ -1,40 +1,51 @@
...
-RUN git clone -b ${VERSION} https://github.com/linkedin/cruise-control.git
-RUN cd cruise-control && ./gradlew jar copyDependantLibs
+RUN \
+  set -xe; \
+  apt-get update -qq \
+  && apt-get install -qq --no-install-recommends \
+    git ca-certificates
+RUN \
+  set -xe; \
+  git clone --branch ${VERSION} --single-branch https://github.com/linkedin/cruise-control.git \
```
Comment (schultheiszn) [inline Dockerfile:12]: I'd use `--depth 1` instead of `--single-branch`

**Review by schultheiszn — APPROVED** [review]

**Review by schultheiszn — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (chrisgacsal) [reply Dockerfile]: @schultheiszn Agreed. Fixed.

**Pull request merged**